lib/cli/src/commands/compile.rs

```diff
@@ -105,15 +106,26 @@ impl Compile {
 
         #[cfg(feature = "object-file")]
         if engine_type == EngineType::ObjectFile {
-            let artifact: &wasmer_engine_object_file::ObjectFileArtifact =
-                module.artifact().as_ref().downcast_ref().context("Engine type is ObjectFile but could not downcast artifact into ObjectFileArtifact")?;
-            let symbol_registry = artifact.symbol_registry();
-            let metadata_length = artifact.metadata_length();
+            use wasmer_engine_object_file::ObjectFileArtifact;
+
+            let symbol_registry = unsafe {
+                module.query_artifact(|artifact: &ObjectFileArtifact| {
+                    Some(Arc::clone(artifact.symbol_registry()))
+                })
+            };
+            let metadata_length = unsafe {
+                module.query_artifact(|artifact: &ObjectFileArtifact| {
+                    Some(artifact.metadata_length())
+                })
+            };
+
```

Comment:
This can be done in 1 unsafe block or even in 1 query; I have a slight preference for that because `unsafe` should be a marker for failure-prone code but this is not, so reducing the use of keyword `unsafe` by half is a good thing here.